File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum=INT_MIN,maxTmp=0;
        for(int i=0;i<nums.size();i++){
            maxTmp=maxTmp+nums[i];
            if(maxTmp>=maxSum){
                maxSum=maxTmp;
            }
            if(maxTmp<0){
                maxTmp=0;
            }
        }
        return maxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int maxsum = nums[0];
        int lastmax = nums[0];

        for (int i = 1; i < nums.Length; i++) {
            // If extending the current subarray is greater than the current element itself,
            // then extend the subarray; otherwise, start a new subarray.
            lastmax = Math.Max(lastmax + nums[i], nums[i]);

            // Update the global maximum if the current subarray sum is greater.
            maxsum = Math.Max(maxsum, lastmax);
        }

        return maxsum;
    }
}
